neo4j/io/_bolt4.py

```diff
@@ -389,11 +376,20 @@ def fetch_message(self):
 
         return len(details), 1
 
-    def _set_defunct(self, error=None):
-        direct_driver = isinstance(self.pool, BoltPool)
+    def _set_defunct_read(self, error=None):
+        message = "Failed to read from defunct connection {!r} ({!r})".format(
```

Comment:
Best to correct this to use lazy interpolation through the logging framework, e.g. `log.info("Thing %r failed", thing)`.

Reply:
Actually, since this message is also used as argument to the error that `_set_defunct` raises, it will need to be interpolated sooner or later anyway.